modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -0,0 +1,1179 @@
+            } else {
+                // Call diagonal
+                preprocessed = Diagonal(
+                    !preprocessed.empty() ? preprocessed : inputs[inputIter],
+                    subscriptIndicesToInputIndex[subscriptIndex],
+                    dimIndexInIreprocessedInput);
+            }
+            ++dimIndexInOriginalInput;
+        }
+
+        std::vector<size_t> permutation;
+        for(auto& d : subscriptIndicesToInputIndex)
+        {
+            if (d != -1)
+                permutation.emplace_back(d);
+        }
+
+        if (IsTransposeRequired(
+            !preprocessed.empty() ? preprocessed.size.dims() : inputs[inputIter].size.dims(),
+            permutation))
+        {
+            // call transpose
+            preprocessed = Transpose(
+                !preprocessed.empty() ? preprocessed : inputs[inputIter],
+                !preprocessed.empty() ? shape(preprocessed) : shape(inputs[inputIter]),
+                permutation);
+        }
+
+        if (!preprocessed.empty())
+        {
+            preprocessed = preprocessed.reshape(1, homogenizedInputDims_.size(), homogenizedInputDims_.data());
+        }
+
+        preProcessedInputs.emplace_back(preprocessed);
+        homogenizedInputDims.emplace_back(homogenizedInputDims_);
+        ++inputIter;
+    }
+}
+
+void LayerEinsumImpl::parseEquation(String equation)
+{
+    // remove white spaces in the copy
+    equation.erase(std::remove_if(equation.begin(), equation.end(), ::isspace), equation.end());
+
+    // check if '->' - the output subscript label is present in the equation;
+    std::size_t arrow_idx = equation.find("->");
+    if (arrow_idx != std::string::npos)
+    {
+        // split left and righ hand sides of the equation
+        lhs_eq = equation.substr(0, arrow_idx);
+        rhs_eq = equation.substr(arrow_idx + 2);
+        explicitEquation = true;
+    } else {
+        lhs_eq = equation;
+    }
+
+    // split lhs_eq by ',' - comma and put all created token - splits
+    // into lhs_eq_tokens vector
+    String token, comma = ",";
+    size_t idx = 0;
+    size_t start = 0;
+    while(idx != String::npos){
+        idx = lhs_eq.find(comma, start);
+        token = lhs_eq.substr(start, idx - start);
+        start = idx + comma.length();
+        lhs_eq_tokens.emplace_back(token);
+    }
+}
+
+
+void LayerEinsumImpl::calculateOutputShape()
+{
+
+
+    // Traverse through each of the subscript labels within the output subscript.
+    bool middleOfEllipsis = false;
+    // int64_t ellipsisCharCount = 0;
+
+    subscriptIndicesToOutputIndices.resize(numLetterIndices, -1);
+
+    std::array<int, numOfLetters> outputLetterToCount;
+    outputLetterToCount.fill(0);
+
+    int outputDimCounter = 0;
+    for (auto letter : rhs_eq)
+    {
+        if(letter == '.')
+        {
+            CV_Error(Error::StsNotImplemented, "Ellipsis are not supported yet");
+        } else {
```

Comment:
You can use `CV_CheckNE(letter, static_cast<int>('.'), "Ellipsis are not supported yet")` to eliminate the if conditioning here.

Reply:
@Abdurrahheem, still relevant